### PR TITLE
fix  user image previewer

### PR DESCRIPTION
**Pull request by tri-et**

### CODE REVIEW

File: lib/ui/widgets/user_image_previewer.dart [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  final TextEditingController photoUrlController;
+  final bool loadingState;
+  UserImagePreviewer({@required this.photoUrlController, @required this.loadingState});
+
+  @override
+  _UserImagePreviewerState createState() => _UserImagePreviewerState();
+}
+
+class _UserImagePreviewerState extends State<UserImagePreviewer> {
+  final photoUrlFocusNode = FocusNode();
```
Comment (hemvanh) [inline lib/ui/widgets/user_image_previewer.dart:21]: needs to dispose

File: lib/ui/widgets/user_settings.dart [hunk 2]
```diff
@@ -1,13 +1,19 @@
...
-class UserSettings extends StatelessWidget {
+class UserSettings extends StatefulWidget {
+  @override
+  _UserSettingsState createState() => _UserSettingsState();
+}
+
+class _UserSettingsState extends State<UserSettings> {
   final displayNameController = TextEditingController();
   final photoUrlController = TextEditingController();
+  final photoUrlFocusNode = FocusNode();
```
Comment (hemvanh) [inline lib/ui/widgets/user_settings.dart]: remove photoUrlFocusNode here, and need to dispose controllers

File: lib/ui/widgets/user_settings.dart [hunk 3]
```diff
@@ -35,18 +41,9 @@ class UserSettings extends StatelessWidget {
...
-            ),
-            SizedBox(
-              height: 20,
-            ),
-            Container(
-              width: 120,
-              height: 120,
-              child: notifier.busy ? _imageLoading : _userImage(photoUrlController.text),
+            UserImagePreviewer(
+              photoUrlController: photoUrlController,
```
Comment (hemvanh) [inline lib/ui/widgets/user_settings.dart]: photoUrlController -> can just pass the text over ?

File: lib/core/notifiers/user_image_previewer_notifier.dart [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import 'package:ata/core/notifiers/base_notifier.dart';
+
+class UserImagePreviewerNotifier extends BaseNotifier {
+  String photoUrl;
+
+  void userImagePreviewer(String url) {
```
Comment (hemvanh) [inline lib/core/notifiers/user_image_previewer_notifier.dart]: userImagePreviewer -> updatePhotoUrl

File: lib/core/notifiers/user_settings_notifier.dart [hunk 5]
```diff
@@ -41,4 +41,11 @@ class UserSettingsNotifier extends BaseNotifier {
       },
     );
   }
+
+  void userImagePreviewer(String displayNameLocal, String photoUrlLocal) {
```
Comment (hemvanh) [inline lib/core/notifiers/user_settings_notifier.dart]: remove this ?

**Review by hemvanh — CHANGES_REQUESTED** [review]

File: lib/ui/widgets/user_settings.dart (hunk 3 above)
Comment (tri-et) [reply lib/ui/widgets/user_settings.dart]: can not

File: lib/ui/widgets/user_image_previewer.dart [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+  @override
+  void dispose() {
+    print("disposeddd");
```
Comment (hemvanh) [inline lib/ui/widgets/user_image_previewer.dart]: remove debug print

File: lib/ui/widgets/user_image_previewer.dart [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+    photoUrlFocusNode.dispose();
+    super.dispose();
```
Comment (hemvanh) [inline lib/ui/widgets/user_image_previewer.dart:26]: super.dispose() should be called first

File: lib/ui/widgets/user_settings.dart (hunk 3 above)
Comment (hemvanh) [reply lib/ui/widgets/user_settings.dart]: set text for text field we must use controller, won't have another way to set text

**Review by hemvanh — CHANGES_REQUESTED** [review]

**Review by hemvanh — APPROVED** [review]

**Pull request merged**